src/generic/toolbkg.cpp

```diff
@@ -302,28 +305,42 @@ bool wxToolbook::InsertPage(size_t n,
     m_maxBitmapSize.x = wxMax(bitmap.GetWidth(), m_maxBitmapSize.x);
     m_maxBitmapSize.y = wxMax(bitmap.GetHeight(), m_maxBitmapSize.y);
 
+    int toolId = page->GetId();
     GetToolBar()->SetToolBitmapSize(m_maxBitmapSize);
-    GetToolBar()->AddRadioTool(n + 1, text, bitmap, bitmap.ConvertToDisabled(), text);
+    GetToolBar()->InsertTool(n, toolId, text, bitmap, bitmap.ConvertToDisabled(), wxITEM_RADIO);
 
-    if (bSelect)
+    // fix current selection
+    if (m_selection < 0)
```

Comment:
It's more clear to precisely check for the absence of selection:

```suggestion
    if ( m_selection != wxNOT_FOUND )
```